llms/mlx_lm/sample_utils.py

```diff
@@ -0,0 +1,38 @@
+import mlx.core as mx
+
+
+def top_p_filtering(logits: mx.array, top_p: float, temperature: float) -> mx.array:
```

Comment:
I would change the name to `top_p_sampling` rather than filtering, or even `nucleus_sampling`.

Reply:
Yeah, I renamed it to top_p_sampling because I personally feel that "top_p" is more widely used. Let me know if you still think `nucleus_sampling` is better.

Reply:
No it's great! My comment was mostly to use sampling over filtering.